get_global_param.c

```diff
@@ -102,11 +102,41 @@ void ProgramState::initGrid(const std::vector<cell_info_struct>& cells) {
         }
       }
     }
-    global_param.gridNumLatDivisions = ((largestLat - global_param.gridStartLat) / global_param.gridStepLat) + 1;
-    global_param.gridNumLonDivisions = ((largestLon - global_param.gridStartLon) / global_param.gridStepLon) + 1;
+    global_param.gridNumLatDivisions = ((global_param.gridEndLat - global_param.gridStartLat) / global_param.gridStepLat) + 1;
+    global_param.gridNumLonDivisions = ((global_param.gridEndLon - global_param.gridStartLon) / global_param.gridStepLon) + 1;
   }
 }
 
+void ProgramState::initCellMask(const std::vector<cell_info_struct>& cells) {
+
+  char            ErrStr[MAXSTRING];
+	int total_num_cells = global_param.gridNumLatDivisions * global_param.gridNumLonDivisions;
+	valid_cell_mask = new bool [total_num_cells];
```

Comment:
They're not in lock step anyways... `valid_cell_idx` and `cell_idx` [are incremented under different circumstances](https://github.com/pacificclimate/VIC/blob/handle-ignored-cells/WriteOutputAllCells.c#L65). I agree somewhat that having the the dimensionality agree between the two has advantages. But the mapping between a 1d cell array and a 2d grid is already loose and has some dubious assumptions.

For example, what happens if a user provides a cell array that *doesn't* proceed from smallest to largest latitude and from smallest to largest longitude? Remember that (for better or for worse) cells are defined in a big, long user input list by lat/lon with no order implied or assumed. If the user chooses a different cell ordering structure (or even a random order), which there's nothing stopping them from doing, this output code won't get the positions correct.